src/__tests__/aux/lolex.js

```diff
@@ -0,0 +1,110 @@
+// @flow strict-local
+
+import LolexModule from 'lolex';
+
+/*
+ * At present (Jest v24.9.0), Jest does not override Date.now() when using a
+ * fake timer implementation. This means that any timer-based code relying on
+ * Date.now() for throttling (etc.) will be very confused.
+ *
+ * The good news is that Jest is very close to using Lolex internally -- see,
+ * e.g., https://github.com/facebook/jest/pull/7776 -- at which point that
+ * behavior will be available to us via Jest. The bad news, alas, is that it's
+ * not there yet.
+ *
+ * For now, we borrow slices of Jest's planned Lolex-based timer implementation.
+ */
+
+/**
+ * A Lolex-backed implementation of certain relevant Jest functions.
+ *
+ * Carved from the more-complete, not-yet-NPM-available implementation at:
+ * https://github.com/facebook/jest/blob/9279a3a97/packages/jest-fake-timers/src/FakeTimersLolex.ts
+ *
+ * Instantiating one of these will switch Jest over to using Lolex's
+ * `Date.now()` replacement. Calling `.dispose()` on that instantiation will
+ * remove that. (Behavior in the presence of multiple Lolex instances is not
+ * defined. Don't do that.)
+ *
+ * Users of this class are recommended to use Jest's setup and teardown
+ * functions, perhaps as follows:
+ *
+ * ```
+ * describe('description', () => {
+ *   const lolex: Lolex = new Lolex();
+ *
+ *   afterEach(() => { lolex.clearAllTimers(); });
+ *   afterAll(() => { lolex.dispose(); });
+ *
+ *   // ...tests...
+ * });
+ * ```
+ */
+export class Lolex {
+  /** The installed Lolex clock object. (Name also taken from Jest's
+      implementation, for simplicity's sake. */
+  _clock;
+
+  constructor() {
+    this._clock = LolexModule.install();
+
+    // Wrap Lolex's setTimeout and setInterval with additional checks.
+    //
+    // In Node.js, a timeout or interval of greater than INT32_MAX is reduced to
+    // exactly 1. (In browsers, and probably in React, it is instead truncated
+    // to an int32 value.)
```

Comment:
"in React" is a confusing expression here -- React also runs in browsers :-) and I think also in Node. I think you probably mean "React Native"?

Reply:
... yes. :man_facepalming: